### ISSUE TITLE
Polymorphic option should be either deprecated or documented better

### ISSUE THREAD

**Issue by amkisko** [jsonapi-serializer/jsonapi-serializer#140]
It is not clear that `polymorphic` option takes priority over `serializer` option. Also seems that `serializer` option is recommended way for polymorphic relations: https://github.com/jsonapi-serializer/jsonapi-serializer#specifying-a-relationship-serializer

Code state: https://github.com/jsonapi-serializer/jsonapi-serializer/blob/master/lib/fast_jsonapi/relationship.rb#L74 and this behaviour is not documented at all.

My case: I used both `polymorphic` and `serializer` in one statement and for few hours could not realise why it does not call serializer block.

**Comment by stas** [comment]
@amkisko I agree, this is rather confusing. I'm scheduling this for the v3.0.0 which will introduce a rewritten version of it.

Thank you!
